### DISCUSSION TITLE
Logging for trace

### DISCUSSION THREAD

**Discussion by Denisjc** [VsixCommunity/Community.VisualStudio.Toolkit#482] [General]
Do we have a recommended logging process that we can use to trace the activity of the extension?

**Comment by madskristensen** [comment]
We don't. You can use App Insights or any other logging service available for .NET Framework

**Comment by StevenRasmussen** [comment]
I've had success logging to the output window along with logging to a file using Serilog (which has many options for logging including to App Insights as Mads suggested).  Logging to the output window provides great feedback to the user if you have operations that hit web services or other longer running tasks.

**Reply by RobertvanderHulst** [reply]
We also use Serilog. It works great
